test/sagemaker_endpoint_tests/test_endpoint.py

```diff
@@ -0,0 +1,106 @@
+            sagemaker_session=sagemaker_session
+        )
+        
+        predictor = model.deploy(
+            initial_instance_count=1,
+            instance_type=instance_type,
+            endpoint_name=endpoint_name,
+        )
+        LOGGER.info(f"Successfully deployed endpoint: {endpoint_name}.")
+    except Exception as e:
+        LOGGER.error(f"Error while setting up SageMaker Endpoint:{endpoint_name}. {e}")
+    finally:
+        if sagemaker_session:
+            sagemaker_session.delete_endpoint(endpoint_name)
+            LOGGER.info(f"Deleted endpoint: {endpoint_name}.")
+
+
+@pytest.mark.sagemaker_endpoint("pytorch")
+def test_pytorch_endpoint(image_uri, sagemaker_region, instance_type, framework_version, py_version, pt_model_dir, pt_model_script):
+    prefix = 'sagemaker-pytorch-serving/models'
+    sagemaker_session = utils.get_sagemaker_session(sagemaker_region)
+    endpoint_name = utils.create_endpoint_name("sagemaker-pytorch-serving")
+    instance_type = instance_type or "ml.g4dn.xlarge"
+    try:
+        model_data = sagemaker_session.upload_data(path=pt_model_dir, key_prefix=prefix)
+        model = PyTorchModel(
+            model_data=model_data,
+            role="SageMakerRole",
+            entry_point=pt_model_script,
+            framework_version=framework_version,
+            py_version=py_version,
+            image_uri=image_uri,
+            sagemaker_session=sagemaker_session
+        )
+
+        predictor = model.deploy(
+            initial_instance_count=1,
+            instance_type=instance_type,
+            endpoint_name=endpoint_name,
+        )
+        LOGGER.info(f"Successfully deployed endpoint: {endpoint_name}.")
+    except Exception as e:
+        LOGGER.error(f"Error while setting up SageMaker Endpoint:{endpoint_name}. {e}")
+    finally:
+        if sagemaker_session:
+            sagemaker_session.delete_endpoint(endpoint_name)
+            LOGGER.info(f"Deleted endpoint: {endpoint_name}.")
+
+
+@pytest.mark.sagemaker_endpoint("tensorflow")
+def test_tensorflow_endpoint(image_uri, sagemaker_region, instance_type, framework_version, tf_model_dir, tf_model_script):
+
+    prefix = 'sagemaker-tensorflow-serving/models'
+    sagemaker_session = utils.get_sagemaker_session(sagemaker_region)
+    endpoint_name = utils.create_endpoint_name("sagemaker-tensorflow-serving")
+    instance_type = instance_type or "ml.g4dn.xlarge"
+    try:
+        model_data = sagemaker_session.upload_data(path=tf_model_dir, key_prefix=prefix)
+        model = TensorFlowModel(
+            model_data=model_data,
+            role="SageMakerRole",
+            entry_point=tf_model_script,
+            framework_version=framework_version,
+            image_uri=image_uri,
+            sagemaker_session=sagemaker_session
+        )
+
+        predictor = model.deploy(
+            initial_instance_count=1,
+            instance_type=instance_type,
+            endpoint_name=endpoint_name,
+        )
```

Comment:
Please add a timeout functionality for these test endpoints, because we do not want incomplete/failed/successful endpoints to remain up after the tests are complete.

Refer: https://github.com/aws/deep-learning-containers/blob/32bd3dfc242782aace3005e2d0ca8c43a2962068/test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_hosting.py#L49

Reply:
Adding the timeout functionality is leading to SageMaker endpoint not being deleted at the end of the tests. The timeout functionality will be more applicable in an instance where we are actually running an inference job. Using the finally with try and except is ensuring that the endpoint gets taken down once the creation is complete.